amivapi/utils.py

```diff
@@ -18,17 +18,24 @@
 from flask import g
 
 
-def token_urlsafe():
+def token_urlsafe(nbytes=None):
     """Cryptographically random generate a token that can be passed in a URL.
-    The token is created from 256 random bits.
 
     This function is available as secrets.token_urlsafe in python3.6. We can
     remove this function when we drop python3.5 support.
 
+    Args:
+        nbytes: Number of random bytes used to generate the token. Note that
+        this is not the resulting length of the token, just the amount of
+        randomness.
+
     Returns:
         str: A random string containing only urlsafe characters.
     """
-    return b64encode(urandom(32)).decode("utf-8").replace("+", "-").replace(
+    if nbytes is None:
+        nbytes = 16
```

Comment:
I'm curious: could we not just use `nbytes=16` in the function definition? Using `None` as a default parmeter is only needed for mutable objects as defaults -- or is there another reason I am not aware of?